README.md

```diff
@@ -34,3 +34,16 @@ Now you can run tests, using forge:
 ```bash
 forge test
 ```
+
+## Specifications
+
+### The pending root is not a queue
+The pending root does not have a queue mechanism. Therefore, pushing a root to the pending root as a root updater in a distribution with a timelock will essentially erase the previous root. This means that a compromised root updater can always suggest a root and reset the timelock of the pending root.
+
+Additionally, if the pending root is ready to be accepted but a root updater suggests a new root at the same time, the pending root will be erased and the timelock will restart.
+
+This can lead to infinite loops of pending root if an automation mechanism suggests a root at an interval smaller than the timelock.
+
+This behavior is acknowledged. When designing a strategy on top of the URD, you must ensure that the epoch interval at which you update the root is longer than the timelock. Define these parameters accordingly.
```

Comment:
Or you can revoke the update. I think we should say that this solutions is available as well